src/main/java/cloud/orbit/actors/extensions/metrics/dropwizard/OrbitActorExtension.java

```diff
@@ -137,40 +141,53 @@ public void onRead(HandlerContext ctx, Object message)
     {
         //actor counter per type and node
         String actorCounterKey = getActorCounterMetricsKey(RemoteReference.getInterfaceClass(actor));
-        Counter counter = MetricsManager.getInstance().getRegistry().counter(actorCounterKey);
-        counter.dec();
+        MetricsManager.getInstance().getRegistry().counter(actorCounterKey).dec();
 
-        //actor lifespan
-        String startTimeKey = getActorStartTimeKey(actor);
+        //actor deactivation
+        String actorDeactivationTimerKey = getDeactorActivationTimerKey(actor);
+        Timer.Context deactivationTimer = actorDeactivationTimers.remove(actorDeactivationTimerKey);
+        if (null != deactivationTimer)
+        {
+            deactivationTimer.stop();
+        }
 
-        Long startTime = actorStartTimes.get(startTimeKey);
-        if (null != startTime)
+        //Actor lifespan
+        String actorLifespanTimerKey = getActorLifespanTimerKey(actor);
+        String actorTypeLifespanMetricsKey = getActorLifeSpanMetricsKey(RemoteReference.getInterfaceClass(actor));
+        Timer.Context lifespanTimer = actorLifeSpanTimers.remove(actorLifespanTimerKey);
+        if (null != lifespanTimer)
         {
-            actorStartTimes.remove(startTimeKey);
-
-            String actorLifeSpanHistogramKey = getActorLifeSpanHistogramMetricsKey(RemoteReference.getInterfaceClass(actor));
-            Histogram histogram = actorLifeSpanHistograms.get(actorLifeSpanHistogramKey);
-            if (null != histogram)
-            {
-                histogram.update(System.currentTimeMillis() - startTime);
-            }
+            lifespanTimer.stop();
         }
+
         return Task.done();
     }
 
-    public static String getActorStartTimeKey(final AbstractActor<?> actor)
+    public static String getActorActivationTimerKey(final AbstractActor<?> actor)
```

Comment:
We shouldn't need 3 timer keys. As they are being stored in different collections it would be more effective to have just one key for actors.%s.%s.timer.